waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/service/SSIKit2WalletService.kt

```diff
@@ -124,11 +127,11 @@ class SSIKit2WalletService(
     }
 
     override suspend fun restoreCredential(id: String): WalletCredential =
-        credentialService.restore(walletId, id) ?: error("Credential not found: $id")
+        credentialService.restore(walletId, id) ?: throw NotFoundException("Credential not found: $id")
 
     override suspend fun getCredential(credentialId: String): WalletCredential =
         credentialService.get(walletId, credentialId)
-            ?: throw IllegalArgumentException("WalletCredential not found for credentialId: $credentialId")
+            ?: throw NotFoundException("WalletCredential not found for credentialId: $credentialId")
```

Comment:
This error message exposes a class and a parameter name that doesn't make sense to the final user. Why not maintain the consistency with other messages just with "Credential not found: $id"?

Reply:
However, I would still change that last message to "Credential with ID $id not found" because, just by looking at the final error message "Credential not found: 98723645453738", it's not clear what this number is about.